### ISSUE TITLE
Migrating docker-compose from 2 to 3,  Error response from daemon: rpc error: code = 3 desc = driver name: if driver is specified name is required

### ISSUE THREAD

**Issue by jonathan-kosgei** [docker/docker#30758]
**Description**

`docker stack deploy` doesn't work as expected on 1.13

**Steps to reproduce the issue:**
1. Run `docker stack deploy` with the following compose file

```
version: '3'
volumes:
  poc:
services:
  redis:
    image: 172.31.25.208:5000/redis
    logging:
        driver: journald
    volumes:
      - poc:/redis
    networks: [uncontainer]

networks:
  uncontainer:
    driver: overlay
    ipam:
      config:
        - subnet: 100.64.0.0/16
```

**Describe the results you received:**
Running the above compose file gives
```
root@ip-172-31-25-208:~/docker-setup# docker stack deploy --compose-file=docker-compose.yml uncontainer
Creating network uncontainer_uncontainer
Error response from daemon: rpc error: code = 3 desc = driver name: if driver is specified name is required
```

**Describe the results you expected:**
Successful deploy of my app containers on my swarm cluster

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:      1.13.0
 API version:  1.25
 Go version:   go1.7.3
 Git commit:   49bf474
 Built:        Tue Jan 17 09:58:26 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.0
 API version:  1.25 (minimum version 1.12)
 Go version:   go1.7.3
 Git commit:   49bf474
 Built:        Tue Jan 17 09:58:26 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 31
 Running: 15
 Paused: 0
 Stopped: 16
Images: 116
Server Version: 1.13.0
Storage Driver: overlay2
 Backing Filesystem: xfs
 Supports d_type: true
 Native Overlay Diff: false
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: active
 NodeID: cbtslk7duprzq29i530f4i8y5
 Is Manager: true
 ClusterID: 0i22jyym0uiwvzkf3ehud7kub
 Managers: 1
 Nodes: 2
 Orchestration:
  Task History Retention Limit: 10
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 172.31.25.208
 Manager Addresses:
  172.31.25.208:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 03e5862ec0d8d3b3f750e19fca3ee367e13c090e
runc version: 2f7393a47307a16f8cee44a37b262e8b81021e3e
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-36-generic
Operating System: Ubuntu 16.04.1 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.858 GiB
Name: ip-172-31-25-208
ID: 5TRV:4MMS:OASJ:YC7Q:Y6K4:CO5M:PKMS:MQR3:PH6A:Y2U3:PBSJ:2OGW
Docker Root Dir: /docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Experimental: false
Insecure Registries:
 172.31.25.208:5000
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
AWS

**Comment by vdemeester** [comment]
@jonathan-kosgei This should be fix in 1.13.1, see https://github.com/docker/docker/pull/29826

**Comment by jonathan-kosgei** [comment]
Upgrading to 1.13.1 fixed it thanks

For anyone who comes across this, you need to enable the `testing` repository
```
$ sudo apt-get install software-properties-common
$ sudo add-apt-repository \
       "deb https://apt.dockerproject.org/repo/ \
       ubuntu-$(lsb_release -cs) \
       main testing"
$ sudo apt-get update
$ sudo apt-get -y install docker-engine
```
